features/events/daemon/src/main/java/org/opennms/netmgt/eventd/sink/EventsModule.java

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+/**
+ * @author ms043660 (Malatesh.Sudarshan@cerner.com)
+ */
+package org.opennms.netmgt.eventd.sink;
+
+import java.util.Objects;
+
+import org.opennms.core.ipc.sink.api.AggregationPolicy;
+import org.opennms.core.ipc.sink.api.AsyncPolicy;
+import org.opennms.core.ipc.sink.xml.AbstractXmlSinkModule;
+import org.opennms.netmgt.config.api.EventdConfig;
+import org.opennms.netmgt.events.api.EventsWrapper;
+import org.opennms.netmgt.xml.event.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventsModule extends AbstractXmlSinkModule<EventsWrapper, Log> {
```

Comment:
Since the `Log` implements the `Message` interface, it doesn't look like there's any need for the `EventsWrapper` wrapper.

That said, I think we'd be better off using an `Event` as the initial message rather than a `Log` here. Otherwise we need to create a `Log` every time we need to send a single event.

This won't change anything for your specific use case, but will make it easier to use when dispatching messages directly with the Sink API.

Reply:
Yup i would update to event from log rather than doing to log at sender side its better to reuse and do at receiver under sendnowsync as usual.However we need eventswrapper class because MessageConsumer<S extends Message, T extends Message> uses two Message argument and we cant use Events or Log as first one since both return Events and in aggregate we cant iterate over Events and get single event or we cant hardcode to return first event .Hence EventsWrapper is used.